src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -1158,17 +1157,13 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
         separator = " "s;
       }
 
-      result = ExpressionPtr{
-          std::make_unique<sparqlExpression::GroupConcatExpression>(
-              distinct, std::move(childExpression), std::move(separator))};
+      return makePtr.operator()<sparqlExpression::GroupConcatExpression>(
+          std::move(separator));
     } else {
       AD_CHECK(ad_utility::getLowercase(children[0]->getText()) == "sample");
-      result =
-          ExpressionPtr{std::make_unique<sparqlExpression::SampleExpression>(
-              distinct, std::move(childExpression))};
+      return makePtr.operator()<sparqlExpression::SampleExpression>();
     }
-    result->descriptor() = context->getText();
-    return result;
+    AD_FAIL()  // Should be unreachable.
```

Comment:
If we have the `AD_FAIL()` here, then the last case should be an `else if ` as well instead of `else AD_CHECK`, otherwise it's a bit redundant.